app/src/main/java/com/github/fribourgsdp/radio/SetUtility.kt

```diff
@@ -15,4 +15,13 @@ object SetUtility {
     fun <E> addAllToSet(set : MutableSet<E>, items : Iterable<E>){
         items.forEach{ i : E -> addToSet(set, i)}
     }
+
+    fun <E : Nameable> getNamedFromSet(set : MutableSet<E>, name: String) : E {
+        val filteredSet = set.filter { nameable -> nameable.name == name }
+        if (filteredSet.isEmpty()) {
+            throw NoSuchElementException()
+        } else {
+            return filteredSet[0]
```

Comment:
Are Nameable elements in a set uniquely defined by their name attribute? If so I see no problems here but if not, won't we lose some data by only taking the first elem?

Reply:
hmmm you are right, this implementation will not work with identically named songs of different artists. I will change it next week to a better implementation